packages/e2e-tests/src/specs/dashboard/settings/adminUser/customFonts.js

```diff
@@ -0,0 +1,113 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  withExperimentalFeatures,
+  visitSettings,
+  addCustomFont,
+  removeCustomFont,
+  getFontList,
+  getSelectedFontList,
+  takeSnapshot,
+} from '@web-stories-wp/e2e-test-utils';
+
+const FONT_BASE_URL = `${process.env.WP_BASE_URL}/wp-content/e2e-assets`;
+const OPEN_SANS_CONDENSED_LIGHT = 'Open Sans Condensed Light';
+const OPEN_SANS_CONDENSED_LIGHT_URL = `${FONT_BASE_URL}/OpenSansCondensed-Light.ttf`;
+const OPEN_SANS_CONDENSED_BOLD = 'Open Sans Condensed Bold';
+const OPEN_SANS_CONDENSED_BOLD_URL = `${FONT_BASE_URL}/OpenSansCondensed-Bold.ttf`;
+const OPEN_SANS_CONDENSED_LIGHT_ITALIC_URL = `${FONT_BASE_URL}/OpenSansCondensed-LightItalic.ttf`;
+
+const findByUrl = (arr, val) => {
+  const found = arr.filter((o) => o.url === val);
+  if (found) {
+    return found[0];
+  }
+  return [];
+};
+
+const asyncForEach = async (array, callback) => {
+  for (let index = 0; index < array.length; index++) {
+    // eslint-disable-next-line no-await-in-loop -- async foreach
+    await callback(array[index], index, array);
+  }
+};
+
+const removeAllFonts = async () => {
+  await visitSettings();
+  let fonts = await getFontList();
+
+  if (fonts.length === 0) {
+    return;
+  }
+
+  await asyncForEach(fonts, async (font) => {
+    await removeCustomFont(font.name);
+  });
+
+  await page.waitForTimeout(500);
+  fonts = await getFontList();
+  expect(fonts).toHaveLength(0);
+};
+
+describe('Custom Fonts', () => {
+  withExperimentalFeatures(['customFonts']);
+
+  beforeEach(async () => {
+    await visitSettings();
+    await removeAllFonts();
+  });
+
+  afterEach(async () => {
+    await removeAllFonts();
+  });
+
+  it('should add a fonts and handle keyboard navigation', async () => {
+    await addCustomFont(OPEN_SANS_CONDENSED_LIGHT_URL);
+    await addCustomFont(OPEN_SANS_CONDENSED_BOLD_URL);
+    await addCustomFont(OPEN_SANS_CONDENSED_LIGHT_ITALIC_URL);
+
+    const fonts = await getFontList();
+    const font1 = findByUrl(fonts, OPEN_SANS_CONDENSED_LIGHT_URL);
+    expect(font1.name).toStrictEqual(OPEN_SANS_CONDENSED_LIGHT);
+    const font2 = findByUrl(fonts, OPEN_SANS_CONDENSED_BOLD_URL);
+    expect(font2.name).toStrictEqual(OPEN_SANS_CONDENSED_BOLD);
+
+    await page.keyboard.press('Tab');
+    await page.keyboard.press('ArrowDown');
+    await page.keyboard.press('ArrowDown');
+    await page.keyboard.press('ArrowUp');
+    await takeSnapshot(page, `Select custom font`);
```

Comment:
Do we need a screenshot here?

Reply:
Will remove -- I don't think we need it.